File: frontend/src/models/product.js
export class Product {
  constructor(
    id,
    name,
    description,
    price,
    endingTime,
    paymentDueDate,
    isSavedWithRelease
  ) {
    this.id = id;
    this.name = name;
    this.description = description;
    this.biddingPrice = {
      price,
      endingTime,
      paymentDueDate,
    };
    this.isSavedWithRelease = isSavedWithRelease;
  }
}

File: backend/src/main/java/edu/miu/cs545/biddingproject/backend/domains/Product.java
package edu.miu.cs545.biddingproject.backend.domains;

import edu.miu.cs545.biddingproject.backend.values.BiddingPrice;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@AllArgsConstructor @NoArgsConstructor
@Data @Builder
public class Product {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    @Column(columnDefinition = "text", nullable = true)
    private String description;
    @Embedded
    private BiddingPrice biddingPrice;
    private boolean savedWithRelease = false;
    private double depositAmount;

    @ManyToOne @JoinColumn(name = "seller_id")
    private Seller seller;
}
